client/extensions/woocommerce/state/data-layer/request/index.js

```diff
@@ -0,0 +1,38 @@
+/**
+ * Internal dependencies
+ */
+import request from 'woocommerce/state/sites/request';
+import { setError } from 'woocommerce/state/sites/status/wc-api/actions';
+
+import {
+	WOOCOMMERCE_REQUEST,
+	WOOCOMMERCE_REQUEST_SUCCESS,
+	WOOCOMMERCE_REQUEST_FAILURE,
+} from 'woocommerce/state/action-types';
+
+export function handleRequest( { dispatch }, action ) {
+	const { method, siteId, path, body } = action;
+
+	return request( siteId )[ method ]( path, body )
+		.then( data => {
+			dispatch( {
+				type: WOOCOMMERCE_REQUEST_SUCCESS,
+				action,
+				data,
+			} );
+		} )
+		.catch( error => {
+			// TODO: Maybe phase out usage of this in favor of the failure action?
+			dispatch( setError( siteId, action, { message: error.toString() } ) );
+			dispatch( {
+				type: WOOCOMMERCE_REQUEST_FAILURE,
+				action,
+				error,
+			} );
+		} );
+}
+
+export default {
```

Comment:
Since this exports a reducer, should it be called `reducer.js`? it took me a little figuring out to see if there was something more going on here. Or maybe it's a "handler" not a reducer, so `handler.js`? (and rename the test accordingly, IMO)